### ISSUE TITLE
Yaml Util could not  dump java.util.Map object which value is type of io.kubernetes.client.custom.Quantity

### ISSUE THREAD

**Issue by kaizhaokover** [kubernetes-client/java#404]
#### Basic Information

- Java client API version: 2.0.0

#### Reproduce Steps
 - write code as below
```java
        V1Container container = new V1Container();
        V1ResourceRequirements redisResourceReq = new V1ResourceRequirements();
        Map<String, Quantity> requestQuantity = new HashMap<>();
        requestQuantity.put("memory", new Quantity("1Gi"));
        requestQuantity.put("cpu", new Quantity("0.5"));
        Map<String, Quantity> limitQuantity = new HashMap<>();
        limitQuantity.put("memory", new Quantity("4Gi"));
        limitQuantity.put("cpu", new Quantity("2"));
        redisResourceReq.requests(requestQuantity);
        redisResourceReq.limits(limitQuantity);
        container.resources(redisResourceReq);
        System.out.println(Yaml.getSnakeYaml().dump(container));
```
 - the output shows below
```yaml
resources:
  limits:
    memory: {}
    cpu: {}
  requests:
    memory: {}
    cpu: {}
```
- Expect:
```yaml
resources:
  limits:
    memory: "1Gi"
    cpu: "0.5"
  requests:
    memory: "4Gi"
    cpu: "2"
```

As you can see , the Yaml util does't work as expected !
I have checked [latest Yaml Util class](https://github.com/kubernetesclient/java/blob/master/util/src/main/java/io/kubernetes/client/util/Yaml.java#L440)
This class also doesn't work well .

**Comment by kondapally1989** [comment]
adding something like RepresentQuantity to yaml CustomRepresnter will print values 
```
resources:
  limits:
    memory: 4Gi
    cpu: '2'
  requests:
    memory: 1Gi
    cpu: 500m
```
but it seems, values are loosing original format because of QunatityFormatter 
will try to create a pr  with this

**Comment by kaizhaokover** [comment]
> adding something like RepresentQuantity to yaml CustomRepresnter will print values
> 
> ```
> resources:
>   limits:
>     memory: 4Gi
>     cpu: '2'
>   requests:
>     memory: 1Gi
>     cpu: 500m
> ```
> but it seems, values are loosing original format because of QunatityFormatter
> will try to create a pr with this
I've slove this problem by create a CustomRepresnter !
But I wanna a common yaml dump util.

**Comment by davidxia** [comment]
I created a [failing test](https://github.com/kubernetes-client/java/compare/master...davidxia:patch1?expand=1) for this.

**Comment by davidxia** [comment]
And a PR here https://github.com/kubernetes-client/java/pull/417.

**Comment by brendandburns** [comment]
Fixed by #417
